test/e2e/framework/pod/wait.go

```diff
@@ -637,140 +527,180 @@ func WaitForPodSuccessInNamespaceSlow(ctx context.Context, c clientset.Interface
 // WaitForPodNotFoundInNamespace returns an error if it takes too long for the pod to fully terminate.
 // Unlike `waitForPodTerminatedInNamespace`, the pod's Phase and Reason are ignored. If the pod Get
 // api returns IsNotFound then the wait stops and nil is returned. If the Get api returns an error other
-// than "not found" then that error is returned and the wait stops.
+// than "not found" and that error is final, that error is returned and the wait stops.
 func WaitForPodNotFoundInNamespace(ctx context.Context, c clientset.Interface, podName, ns string, timeout time.Duration) error {
-	var lastPod *v1.Pod
-	err := wait.PollImmediateWithContext(ctx, framework.PollInterval(), timeout, func(ctx context.Context) (bool, error) {
+	err := framework.Gomega().Eventually(ctx, framework.HandleRetry(func(ctx context.Context) (*v1.Pod, error) {
 		pod, err := c.CoreV1().Pods(ns).Get(ctx, podName, metav1.GetOptions{})
 		if apierrors.IsNotFound(err) {
-			return true, nil // done
-		}
-		if err != nil {
-			return handleWaitingAPIError(err, true, "getting pod %s", podIdentifier(ns, podName))
+			return nil, nil
 		}
-		lastPod = pod
-		return false, nil
-	})
-	if err == nil {
-		return nil
+		return pod, err
+	})).WithTimeout(timeout).Should(gomega.BeNil())
+	if err != nil {
+		return fmt.Errorf("expected pod to not be found: %w", err)
 	}
-	if IsTimeout(err) && lastPod != nil {
-		return TimeoutError(fmt.Sprintf("timed out while waiting for pod %s to be Not Found", podIdentifier(ns, podName)),
-			lastPod,
-		)
-	}
-	return maybeTimeoutError(err, "waiting for pod %s not found", podIdentifier(ns, podName))
+	return nil
 }
 
-// WaitForPodToDisappear waits the given timeout duration for the specified pod to disappear.
-func WaitForPodToDisappear(ctx context.Context, c clientset.Interface, ns, podName string, label labels.Selector, interval, timeout time.Duration) error {
-	var lastPod *v1.Pod
-	err := wait.PollImmediateWithContext(ctx, interval, timeout, func(ctx context.Context) (bool, error) {
-		framework.Logf("Waiting for pod %s to disappear", podName)
-		options := metav1.ListOptions{LabelSelector: label.String()}
-		pods, err := c.CoreV1().Pods(ns).List(ctx, options)
+// PodsResponding waits for the pods to response.
+func WaitForPodsResponding(ctx context.Context, c clientset.Interface, ns string, controllerName string, wantName bool, timeout time.Duration, pods *v1.PodList) error {
```

Comment:
This section seems to be not identical to the change in the referenced commit.  I see here the removal of `WaitForPodToDisappear` which matches the referenced commit, and I see the subsequent function below it `PodsResponding`, and in between I see the introduction of a new function `WaitForPodsResponding`. Is that per expectation? This new function seems to use the "namespace, name" sequence rather than the "name,namespace" introduced in the referenced commit.  Is this as intended?

Reply:
I don't see a test case for the new function, but I do see the new function being used, so I expect this is as intended.  Leave this comment open for @pohly double-check.

Reply:
Its in the original commit. close this comment